docs/sources/lids/_DRAFT-Index-Gateway-Sharding.md

```diff
@@ -0,0 +1,97 @@
+---
+title: "XXXX: Index Gateway Sharding"
+description: Loki Improvement Document for index gateway sharding.
+---
+
+# XXXX: Index Gateway Sharding
+
+**Author:** Christian Haudum (christian.haudum@grafana.com)
+
+**Date:** 02/2023
+
+**Sponsor(s):** @chaudum @owen-d
+
+**Type:** Feature
+
+**Status:** Draft
+
+**Related issues/PRs:**
+
+**Thread from [mailing list](https://groups.google.com/forum/#!forum/lokiproject):**
+
+---
+
+## Background
+
+This document tries to come up with a proposal on how to do a better sharding of data on the index gateways so we are able to scale the service horizontally to fulfill the increased need for metadata queries of big tenants.
+
+The index gateway service can be run in "simple mode", where an index gateway instance is responsible for handling, storing and returning requests for all indices for all tenants, or in "ring mode", where an instance is responsible for a subset of tenants instead of all tenants.
+
+On top of that, in order to achieve redundancy as well as spreading load, the index gateway ring uses by default a replication factor of 3.
+
+This means, before an index gateway client makes a request to the index gateway server, it first hashes the tenant ID and then requests a replication set for that hash from the index gateway ring. Due to the fixed replication factor (RF), the replication set contains three server addresses. On every request, a random server from that list is picked to then execute the request on.
+
+## Problem Statement
+
+The current strategy of sharding by tenant ID and having a replication factor fails in the long run, because even when running lots of index gateways, only a maximum of `n` instances could be utilized by a single tenant, where `n` is the value of the configured RF.
```

Comment:
Is the problem that's some tenants see heavier traffic so a relocation factor of three doesn't scale? If so I think it should be mentioned here.

Reply:
That is right. A replication factor of 3 may not be sufficient for big tenants. It can be increased, however, that does not change the sharding itself. It just means that more instances can be utilized. Since RF is a ring property it affects all tenants alike.